[EN-3445] Update foreign born citizenship fields when applicant has a valid US passport

christopherhuii opened the pull request:
## Description
This PR removes foreign born document questions for foreign born applicants with valid US passports.

This PR also updates test case 8. Test case 8 has a valid US Passport. As a result, when foreign born is selected for citizenship status, the form hides the document type questions and only shows the born on military installation question(s). Validates against XSDs and e-QIP.

Updates to Test Cases:
```

// test1
e-QIP-prototype/api/testdata/complete-scenarios/test1.xml validates

bin/submit testdata/complete-scenarios/test1.json
submit: import into e-QIP successful: agencyKey=1530, requestKey=20022438

// test2
e-QIP-prototype/api/testdata/complete-scenarios/test2.xml validates

bin/submit testdata/complete-scenarios/test2.json
submit: import into e-QIP successful: agencyKey=1530, requestKey=20022439

// test3.json
e-QIP-prototype/api/testdata/complete-scenarios/test8.xml validates

bin/submit testdata/complete-scenarios/test3.json
submit: import into e-QIP successful: agencyKey=1530, requestKey=20022437

// test6
e-QIP-prototype/api/pdf/testdata/complete-scenarios/test6.xml validates

bin/submit testdata/complete-scenarios/test6.json
submit: import into e-QIP successful: agencyKey=1530, requestKey=20022440

// test10
e-QIP-prototype/api/testdata/complete-scenarios/test10.xml validates

bin/submit testdata/complete-scenarios/test10.json
submit: import into e-QIP successful: agencyKey=1530, requestKey=20022441
```

## Checklist for Reviewer

- [ ] Review code changes
- [ ] Review changes for Database effects
- [ ] Verify change works in IE browser

More details about this can be found in [docs/review.md](docs/review.md)

ryanhofdotgov commented:
@christopherhuii For all the JSON scenarios that were modified, any chance that you round trip'd them through the front end to ensure they remained valid and no significant differences were created on save? 

Here's the process I'm thinking:
1. Purge db
2. Load up test1.json after manual? edits
3. Sign in, go to section you made edits – do section review. (Expect no validation errors)
4.  Save section.
5. Go to Review and submit validation (Expect no validation errors)
6. Using `form` command line tool, export live scenario to a file, use diff to compare what was originally loaded. (Expect no significant differences).

I'll add some documentation to test-scenarios.md to describe this – we have an outstanding issue to automatically round-trip all the JSON files.

christopherhuii commented:
@ryanhofdotgov I can load up and check all test cases, but I only changed test8.json.

christopherhuii commented:
I did a very quick run through of all the test cases and found a few errors. All are not related to this PR.

test5
`Your history` -> `Where you went to school` not validating

test7
`Information about you` -> `Your contact information`: At least one contact email is required
`Your history` -> `Where you went to school` not validating

suzubara commented on src/components/Section/Citizenship/CitizenshipConnector.jsx:
```diff
@@ -74,6 +76,13 @@ const connectCitizenshipSection = (Component, {
         return {
           ...citizenship.Status,
+          usPassport: foreign.Passport,
+          validator: data => (
```
I don't love this because IMO `connect` should only be used for passing data to a component from Redux. As an alternative I'd propose importing the Redux `state` directly into the CitizenshipValidator and looking at the value of `foreign.Passport` there, similar to how CitizenshipMultipleValidator looks at `application.Settings.formType` (this isn't a great pattern either but it's more in line with what we're doing in other places).

suzubara commented on src/validators/passport.js:
```diff
@@ -7,81 +7,131 @@ const reBook = '^[a-zA-Z0-9]{9}$'
+
+  const hasPassports = HasPassports.value
+
+  return (
```
since these functions all have the same logic:
```
if (hasPassports === 'No') {
  return true
}
```

I think it would be cleaner to check for that inside this fn and return early instead of executing all 4 functions

christopherhuii commented:
@suzubara I've updated the validation for passports and made the updates to the CitizenshipConnector as requested

@ryanhofdotgov I've reverted test8 and added test11 along with comments in `xml_test.go` describing this new test case. Both have been validated against XSDs and e-QIP.

ryanhofdotgov commented on api/templates/citizenship-status.xml:
```diff
@@ -61,6 +61,13 @@
+        <Name>{{text $citizenship.MilitaryBase}}</Name>
+      </MilitaryBase>
+      {{end}}
+      {{if radio $citizenship.CitizenshipStatus | eq "ForeignBorn"}}
```
This looks like it is redundant with the block starting on line 60? i.e., you could remove the end/and if...ForeignBorn.

christopherhuii replied:
I think I fixed it. Let me know if it looks good.

ryanhofdotgov commented on api/xml/xml_test.go:
```diff
@@ -336,6 +336,14 @@ func TestScenario10(t *testing.T) {
+// `test11` is a clone of `test8` with certain revisions.
+// `test11` is for:
+// * Having an applicant that has a valid U.S. Passport
+// * Having an applicant that has ForeignBorn citizenship
```
```suggestion
// * Having an applicant that has ForeignBorn (aka USByBirthOutsideUS) citizenship and has a U.S. Passport
```
Just trying to add some more clarity because this is such an a confusing aspect. The test is for both of those characteristics simultaneously (i.e., you can't move one of those bullet points to a different scenario and retain test coverage). And technically the passport can be expired.

ryanhofdotgov commented on api/xml/xml_test.go:
```diff
@@ -336,6 +336,14 @@ func TestScenario10(t *testing.T) {
 
+// `test11` is a clone of `test8` with certain revisions.
+// `test11` is for:
+// * Having an applicant that has a valid U.S. Passport
```
```suggestion
```

ryanhofdotgov commented:
@suzubara Would you review the frontend portion of this PR? Thanks
